bjoern/server.c

```diff
@@ -141,46 +196,81 @@ ev_timer_ontick(struct ev_loop* mainloop, ev_timer* watcher, const int events)
 }
 #endif
 
+#ifdef WANT_GRACEFUL_SHUTDOWN
+static void
+ev_shutdown_ontick(struct ev_loop* mainloop, ev_timer* watcher, const int events)
+{
+  ThreadInfo *thread_info = (ThreadInfo*)ev_userdata(mainloop);
+  DBG("ev_shutdown_ontick %d", thread_info->server_info->active_connections);
+
+  if (thread_info->server_info->active_connections == 0) {
+    DBG("No more active connections, shutting down");
+    GIL_LOCK(0);
```

Comment:
Why do we need the GIL here?

Reply:
Oh yeah! We definitely don't. I'll remove this.